### PR TITLE
feat(ctb): OutputBisectionGame Devnet Deployment

### PR DESCRIPTION
**Pull request by refcell**
**Description**

Deploys the `OutputBisectionGame` implementation with an alphabet vm backend to devnet and configures game type `254` in the fault dispute game factory to use this implementation.

Fixes https://github.com/ethereum-optimism/client-pod/issues/257

## Summary by CodeRabbit

- **New Features**
  - Introduced a new game type for output bisection within the dispute resolution system.
  - Added configuration options for the genesis block and split depth settings related to the output bisection game.

- **Enhancements**
  - Improved deployment scripts to handle the new output bisection game type.
  - Enhanced logging to provide better visibility during the deployment process.

- **Documentation**
  - Updated internal documentation to reflect new configuration parameters and deployment steps.

### CODE REVIEW

**Comment by refcell** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #8347** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8347?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/8347?utm_source=stack-comment).

File: packages/contracts-bedrock/scripts/Deploy.s.sol [hunk 1]
```diff
@@ -1065,11 +1081,20 @@ contract Deploy is Deployer {
                     _blockOracle: BlockOracle(mustGetAddress("BlockOracle"))
                 })
             );
-
             uint8 rawGameType = GameType.unwrap(_gameType);
+            string memory gameTypeString;
+            if (rawGameType == 0) {
+                gameTypeString = "Cannon";
+            } else if (rawGameType == 254) {
+                gameTypeString = "OutputBisection";
```
Comment (clabby) [inline packages/contracts-bedrock/scripts/Deploy.s.sol]:
```suggestion
                gameTypeString = "OutputBisectionAlphabet";
```

File: packages/contracts-bedrock/scripts/Deploy.s.sol [hunk 2]
```diff
@@ -1065,11 +1081,20 @@ contract Deploy is Deployer {
                     _blockOracle: BlockOracle(mustGetAddress("BlockOracle"))
                 })
             );
-
```
Comment (clabby) [inline packages/contracts-bedrock/scripts/Deploy.s.sol:1060]: ^ Above, we need to switch between the `FaultDisputeGame` and the `OutputBisectionGame`

File: packages/contracts-bedrock/scripts/Deploy.s.sol [hunk 3]
```diff
@@ -1065,18 +1103,22 @@ contract Deploy is Deployer {
...
+            gameTypeString = "Alphabet";
         } else {
-            console.log(
-                "[WARN] DisputeGameFactoryProxy: `FaultDisputeGame` implementation already set for game type: %s",
-                vm.toString(GameType.unwrap(_gameType))
-            );
+            gameTypeString = "Unknown";
         }
+        console.log(
+            "DisputeGameFactoryProxy: set `FaultDisputeGame` implementation (Backend: %s | GameType: %s)",
```
Comment (clabby) [inline packages/contracts-bedrock/scripts/Deploy.s.sol]: tiny nit: template `FaultDisputeGame` and switch based on the game type in this log.

Comment (clabby) [reply packages/contracts-bedrock/scripts/Deploy.s.sol]: er wait, nvm, I think `GameType` is sufficient here, it'll show the difference in the logs. can ignore.

**Review by clabby — APPROVED** [review]
LGTM. Thanks!

We can alter the configuration values as needed while the feature develops.

File: op-bindings/bindings/outputbisectiongame_more.go [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+func init() {
+	if err := json.Unmarshal([]byte(OutputBisectionGameStorageLayoutJSON), OutputBisectionGameStorageLayout); err != nil {
+		panic(err)
+	}
+
+	layouts["OutputBisectionGame"] = OutputBisectionGameStorageLayout
+	deployedBytecodes["OutputBisectionGame"] = OutputBisectionGameDeployedBin
+}
```
[the remark this answers is not in the document]
Comment (clabby) [reply op-bindings/bindings/outputbisectiongame_more.go:25]: These files are autogenerated, but thanks rabbit

**Comment by refcell** [comment]
Tagging @ajsutton as a reviewer so he's aware this lands in trunk

File: packages/contracts-bedrock/deploy-config/devnetL1-template.json [hunk 5]
```diff
@@ -49,6 +49,8 @@
   "faultGameAbsolutePrestate": "0x03c7ae758795765c6664a5d39bf63841c71ff191e9189522bad8ebff5d4eca98",
   "faultGameMaxDepth": 30,
   "faultGameMaxDuration": 1200,
```
Comment (tynes) [inline packages/contracts-bedrock/deploy-config/devnetL1-template.json:51]: we will need to add these to the other files too, otherwise they will break

Comment (refcell) [reply packages/contracts-bedrock/deploy-config/devnetL1-template.json:51]: I don't believe that's the case - they've been there this whole time?

Comment (tynes) [reply packages/contracts-bedrock/deploy-config/devnetL1-template.json:51]: the new fields in the json i mean

**Review by tynes — APPROVED** [review]

**Pull request merged**